jcl/src/java.base/share/classes/java/lang/Throwable.java

```diff
@@ -489,31 +501,60 @@ private void readObject(ObjectInputStream s)
  * 			number of indents (\t) to be printed
  * @param	suppressed	
  * 			if this is an exception suppressed
+ * @param	exceptionChainSet
+ * 			set of exceptions in the exception chain
  * 
  * @return	an array of stack trace elements printed 
  * 
  */
 private StackTraceElement[] printStackTrace(
-		Appendable err, StackTraceElement[] parentStack, int indents, boolean suppressed) {
+		Appendable err, StackTraceElement[] parentStack, int indents, boolean suppressed, Set<Throwable> exceptionChainSet) {
 	/*[PR 120593] CDC 1.0 and CDC 1.1 TCK fails in java.lang.. Exception tests */
 	if (err == null) throw new NullPointerException();
 	StackTraceElement[] stack;
 	boolean outOfMemory = this instanceof OutOfMemoryError;
+	if ((exceptionChainSet != null) && (exceptionChainSet.contains(this))) {
+		if (!outOfMemory) {
+			try {
+				appendTo(err, "\t[CIRCULAR REFERENCE:" + toString() + "]", 0); //$NON-NLS-1$
+			} catch(OutOfMemoryError e) {
+				outOfMemory = true;
+			}
+		}
+		if (outOfMemory) {
+			appendTo(err, "\t[CIRCULAR REFERENCE:"); //$NON-NLS-1$
+			try {
+				appendTo(err, getClass().getName());
+			} catch (OutOfMemoryError e) {
```

Comment:
For consistency, there shouldn't be a space after `catch`.

Reply:
Done.